static/js/stream_topic_history.js

```diff
@@ -65,6 +65,11 @@ exports.per_stream_history = function (stream_id) {
 
         message_id = parseInt(message_id, 10);
 
+        const max_message_id = self.get_max_message_id();
+        if (message_id > max_message_id) {
+            max_message_ids.set(stream_id, message_id);
+        }
```

Comment:
OK, this doesn't make any sense.  The data structure here is for a single stream, so there's no reason it should be a dictionary, rather than just a number.